src/isl/mod.rs

```diff
@@ -337,4 +338,135 @@ mod isl_tests {
         // determine that the range is created with an error for an invalid range
         assert_eq!(range.is_err(), true);
     }
+
+    #[rstest(
+        range,
+        valid_values,
+        invalid_values,
+        case::int_range(
+            load_range(
+            r#"
+                range::[0, 10]
+            "#
+            ),
+            vec![5.into(), 0.into(), 10.into()],
+            vec![(-5).into(), 11.into()]
+        ),
+        case::int_range_with_min(
+            load_range(
+            r#"
+                range::[min, 10]
+            "#
+            ),
+            vec![5.into(), (-5).into(), 0.into()],
```

Comment:
suggestion—it would be good to also add valid test cases for `0e0`, `0d0`, `-0e0`, `-0d0`, `5d0`, `5e0`, `10e0`, and `10d0`.